Fix: DefaultRealmModule conflict

rawmain opened the pull request:
**This commit fixes build issues when DGC-SDK is integrated into an app project, that uses Realm with default settings for other data management.**

-----------------

### **Issue Description :**

Realm's default behavior is indeed to automatically create a RealmModule called DefaultRealmModule which contains all classes extending RealmObject in a project. This module is automatically known by Realm.

Such default behavior is problematic when combining a library project & an app project that both use Realm with default settings.

This is because the DefaultRealmModule will be created for both the library project and the app project, which will cause the project to fail because of duplicate class definition errors.

- Issue References - https://github.com/ministero-salute/it-dgc-verificac19-sdk-android/issues/90

-----------------

### **Fix details :**

It avoids the DefaultRealmModule conflict by setting an explicit library module for DGC-SDK (_VerificaC19sdkLibraryModule_).

This disables the generation of the DefaultRealmModule for the library project and allows the library to be included in the app project that also uses Realm.

- OK build of VerificaC19 whitelabel-app = built-release 1.2.1 using this edited version of DGC-SDK 1.1.1 is available [HERE](https://github.com/VC19bkp/it-dgc-verificac19-sdk-android-BKP/releases/tag/1.1.1-FIX-REALM).

-----------------

rawmain committed:
Fix - DefaultRealmModule conflict

This commit fixes build issues when DGC-SDK is integrated into an app project, that uses Realm with default settings for other data management.

-----------------

Details :
Realms default behavior is indeed to automatically create a RealmModule called DefaultRealmModule which contains all classes extending RealmObject in a project. This module is automatically known by Realm.

Such default behavior is problematic when combining a library project & an app project that both use Realm with default settings. This is because the DefaultRealmModule will be created for both the library project and the app project, which will cause the project to fail because of duplicate class definition errors.

-----------------

Fix description :
It avoids the DefaultRealmModule conflict by setting an explicit library module for DGC-SDK (VerificaC19sdkLibraryModule). This disables the generation of the DefaultRealmModule for the library project and allows the library to be included in the app project that also uses Realm.

Issue Reference :
https://github.com/ministero-salute/it-dgc-verificac19-sdk-android/issues/90

albertocappellina commented on sdk/src/main/java/it/ministerodellasalute/verificaC19sdk/data/VerifierRepositoryImpl.kt:
```diff
@@ -326,9 +322,11 @@ class VerifierRepositoryImpl @Inject constructor(
-        val config =
-            RealmConfiguration.Builder().name(REALM_NAME).allowQueriesOnUiThread(true)
-                .build()
+        val config = RealmConfiguration.Builder()
```
Don't want to bother but this RealmConfiguration.Builder block should be extracted and moved to some unique point for not to repeat code and, in case of migration, you don't have to search and replace in various places

rawmain replied:
Anyway out of this PR's scope (_"keep it simple" fix_). If/when there would be a required migration owing to any RevokedPass object schema changes, the code would be obviously re-arranged.

rawmain committed:
Merge branch 'ministero-salute:develop' into develop

Kaizen-7 commented on sdk/src/main/java/it/ministerodellasalute/verificaC19sdk/data/local/RevokedPass.kt:
```diff
@@ -25,6 +25,10 @@ package it.ministerodellasalute.verificaC19sdk.data.local
\ No newline at end of file
+open class RevokedPass(@PrimaryKey var hashedUVCI: String = "") : RealmObject()
+
+@RealmModule(library = true, classes = [RevokedPass::class])
```
thanks @rawmain for your support, what do you think about moving the RealmModule declaration outside the RevokedPass.kt class, maybe the code looks clearer?

rawmain committed:
VerificaC19Sdk RealmModule rearrangement

- Renamed class VerificaC19sdkLibraryModule -> VerificaC19sdkRealmModule

- Moved out the RealmModule declaration outside the RevokePass class

- Reverted data/local import wildcard in VerificationViweModel & VerifierRepositoryImpl

rawmain committed:
Merge branch 'ministero-salute:develop' into develop

astagi commented:
@Light2288 I see you approved this PR! can I proceed with merge after CI response?

Light2288 commented:
> @Light2288 I see you approved this PR! can I proceed with merge after CI response?

@astagi yes @Kaizen-7 and me review the pr and it is ok (and very useful indeed). So thank you @rawmain for your contribution and yes @astagi you can proceed to merge it
